doozerlib/cli/detect_embargo_cli.py

```diff
@@ -0,0 +1,175 @@
+@click.option("--yaml", "as_yaml", is_flag=True,
+              help="Print out the result as YAML format.")
+@click.option("--json", "as_json", is_flag=True,
+              help="Print out the result as JSON format.")
+@pass_runtime
+def detect_embargo_cli(runtime, nvrs, pullspecs, release_payload, tags, exclude_tags, kind, as_yaml, as_json):
+    """ Detect embargoed fixes.
+
+    If neither --yaml or --json is given, this program will exit with status 0 if true, or with status 2 if not.
+    Errors are signaled by a non-zero status that is not 2.
+
+    Examples:
+
+    - Check whether one or more brew tags has a package with .p1 and the ability to exclude nvrs which have already shipped.
+
+        doozer detect-embargo --kind rpm --tag rhaos-4.5-rhel-7-candidate:<optional-koji-event#> ... --exclude-tag rhaos-4.5-rhel-7:<optional-koji-event#>
+
+    - Check whether one or more arbitrary brew images has a "release" label with .p1
+
+        doozer detect-embargo --pullspec <image pullspec> ...
+
+    - Check whether one or more nvrs has the .p1
+
+        doozer detect-embargo --nvr <nvr> ...
+
+    - Check whether a release payload contains an image with the release containing .p1
+
+        doozer detect-embargo --release-payload <pullspec> ...
+    """
+    # Enusre only one of --nvr, --pullspec, --release-payload, or --tag is specified
+    n_options = 0
+    if nvrs:
+        n_options += 1
+        nvrs = [NVR.parse(nvr) for nvr in nvrs]  # Tuple[str] -> List[NVR]
+    if pullspecs:
+        n_options += 1
+    if release_payload:
+        n_options += 1
+    if tags:
+        n_options += 1
+    if n_options != 1:
+        raise click.BadParameter("You must use one of --nvr, --pullspec, --release-payload, or --tag.")
+    if exclude_tags and not tags:
+        raise click.BadParameter("--exclude-tag must be used with --tag.")
+    # Enusre only one of --yaml or --json is specified
+    if as_yaml and as_json:
+        raise click.BadParameter("Use one of --yaml or --json.")
+
+    runtime.initialize(no_group=not tags, clone_distgits=False, clone_source=False)
+    logger = runtime.logger
+
+    if release_payload:
+        logger.info(f"Fetching image references from release payload {release_payload}...")
+        pullspecs = list(get_image_pullspecs_from_release_payload(release_payload))
+
+    if pullspecs:
+        logger.info(f"Fetching manifests for {len(pullspecs)} images...")
+        jobs = runtime.parallel_exec(lambda pullspec, terminate_event: get_nvr_by_pullspec(pullspec), pullspecs,
+                                     min(len(pullspecs), multiprocessing.cpu_count() * 4, 32))
+        nvrs = jobs.get()
+
+    if tags:
+        with runtime.shared_koji_client_session() as brew_session:
+            build_type = None if kind == "all" else kind
+            logger.info(f"Fetching builds from Brew tags {tags + exclude_tags}...")
+            tasks = []
+            with brew_session.multicall(strict=True) as m:
```

Comment:
This approach may be slightly faster by using a single multicall, but I would prefer readability here. e.g.
```python
def get_tag_nvrs(tag_event_list):  # just a local closure
    with brew_session... as m:
        for tag, event in [unpack_tag_event(t) for t in tag_event_list]:
     ....
included_nvrs = get_tag_nvrs(tags)
excluded_nvrs = get_tag_nvrs(excluded_tags)
```